### PR TITLE
Add metering of local variables

### PR DESCRIPTION
**Pull request by camilbancioiu**
Added the `feed_local()` method to `FunctionMiddleware`, `GenericFunctionMiddleware`, and the `Metering` middleware. Arwen must now supply an extra gas cost, for allocating a single local variable.

Also added a configurable allowance of unmetered local variables.

Tests for this change will be added to Arwen.

### CODE REVIEW

**Review by sasurobert — CHANGES_REQUESTED** [review]
Check and add the difference of gas cost for the current contracts. Adding this as it is right now it is not backward compatible. Might add these extra local variables gas cost only after 100 variables were allocated.

File: lib/middleware-common/src/metering.rs [hunk 1]
```diff
@@ -111,7 +116,24 @@ impl<'q> FunctionMiddleware for Metering<'q> {
...
+        Ok(())
+    }
+
+    fn feed_local(
+        &mut self,
+        _ty: WpType,
+        n: usize,
+        _loc: u32,
+    ) -> Result<(), Self::Error>{
+        println!("current local: {}", n);
```
Comment (sasurobert) [inline lib/middleware-common/src/metering.rs:131]: why println here ? no need

**Review by sasurobert — APPROVED** [review]

**Review by sasurobert — CHANGES_REQUESTED** [review]
no need for println

File: lib/middleware-common/src/metering.rs (hunk 1 above)
Comment (camilbancioiu) [reply lib/middleware-common/src/metering.rs:131]: Accidentally forgotten. Will remove.

**Review by sasurobert — APPROVED** [review]

**Pull request merged**